app/assets/javascripts/discourse-common/addon/resolver.js

```diff
@@ -85,138 +66,166 @@ export function buildResolver(baseName) {
+      ) {
+        let corrected;
+        // This should only apply when there's a dot or slash in the name
+        if (split[1].includes(".") || split[1].includes("/")) {
+          // Check to see if the dasherized version exists. If it does we want to
+          // normalize to that eagerly so the normalized versions of the dotted/slashed and
+          // dotless/slashless match.
+          const dashed = dasherize(split[1].replace(/[\.\/]/g, "-"));
+          const appBase = `${baseName}/${type}s/`;
+          const adminBase = `admin/${type}s/`;
+          const wizardBase = "wizard/" + split[0] + "s/";
+          if (
+            requirejs.entries[appBase + dashed] ||
+            requirejs.entries[adminBase + dashed] ||
+            requirejs.entries[adminBase + dashed.replace(/^admin[-]/, "")] ||
+            requirejs.entries[
+              adminBase + dashed.replace(/^admin[-]/, "").replace(/-/g, "_")
+            ] ||
+            requirejs.entries[wizardBase + dashed] ||
+            requirejs.entries[wizardBase + dashed.replace(/^wizard[-]/, "")] ||
+            requirejs.entries[
+              wizardBase + dashed.replace(/^wizard[-]/, "").replace(/-/g, "_")
+            ]
+          ) {
+            corrected = type + ":" + dashed;
+          }
         }
 
-        // Try with dashes instead of slashes
-        dashed = dasherize(split[1].replace(/\./g, "-"));
-        if (
-          requirejs.entries[appBase + dashed] ||
-          requirejs.entries[adminBase + dashed] ||
-          requirejs.entries[wizardBase + dashed]
-        ) {
-          return split[0] + ":" + dashed;
+        let wasCorrected = false;
+        if (corrected && corrected !== normalized) {
+          this._uncorrectedValues[fullName] = normalized;
+          normalized = corrected;
+          wasCorrected = true;
         }
-      }
-      return this._super(fullName);
-    },
-
-    customResolve(parsedName) {
-      // If we end with the name we want, use it. This allows us to define components within plugins.
-      const suffix = parsedName.type + "s/" + parsedName.fullNameWithoutType,
-        dashed = dasherize(suffix),
-        moduleName = lookupModuleBySuffix(dashed);
-
-      let module;
-      if (moduleName) {
-        module = requirejs(moduleName, null, null, true /* force sync */);
-        if (module && module["default"]) {
-          module = module["default"];
+
+        // Check if we have any other values that normalized to the same thing. In the future,
+        // when we remove this code, these will no longer work as expected.
+        let match = Object.entries(this._normalizeCache).find(
+          ([key, value]) => {
+            return (
+              value === normalized &&
+              (wasCorrected || this._uncorrectedValues[key])
+            );
+          }
+        );
+        if (match) {
+          // This is set up this way in order to fully strip from prod builds.
+          deprecated(
```

Comment:
`discourse-common/lib/deprecated` is a discourse-specific implementation, and so I don't think it will be stripped from production builds.

Theme developers (and also plugin developers, to a certain extent) develop against production builds, so we deliberately print deprecation messages in prod. Our `deprecated` implementation will automatically prefix the deprecation notice with the name of the theme/plugin.

Reply:
Ok, I guess I'll just change the comment there.